docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+
+This architecture poses a problem for analyzers. It is not possible to easily join the data from the same session between daemon and main application. The session ids are different due to these being two different Glean instances.
+
+For more information on this topic, refer to [Mozilla VPN Telemetry Refactor](https://docs.google.com/document/d/1jyNZ_g_cUpZZsEr2hYwnwZgkxlvqTmoFUJKp_LAOPts/edit#heading=h.xararkvsnpss).
+
+## Proposed Solution
+
+> **Note** Honestly, I think for this proposed solution it's just easier to look at the pseudo-code in Appendix 1.
+
+A new metric will be added to both the daemonsession and the session pings: shared_session_id. This metric will have lifetime ping i.e. it is cleared from storage whenever a ping containing it is submitted. Thus it needs to be set again every time a new session/daemonsession ping is submitted.
+
+Both daemon and main application will store the value of this metric in their local storages for synchronization purposes. More about this in the sections below.
+
+A new reason will be added to the session pings “flush_init”. As better described in the sections below, the processes will be able to identify that either the VPN tunnel or main application did not gracefully hit the deactivation signal i.e. were force killed. The “flush_init” ping will be used to flush any data remaining from the previous session.
+
+There will be four key triggers that will handle the shared_session_id: initialization, on VPN activation, ping timer and on VPN deactivation.
+
+### ON VPN ACTIVATION
+
+#### MAIN APPLICATION
+
+In case the VPN is started from the main application, a session id will be created and shared through the activation signal with the daemon. The generated UUID will be stored in the main application local storage and the session ping will be submitted.
+
+#### DAEMON
+
+If the VPN is being activated through the main application, a session id will have been passed through the activation signal. Otherwise the Daemon will generate this value itself. The UUID will be stored in the daemon local storage and the session ping will be submitted.
+
+### ON PING TIMER
+
+#### MAIN APPLICATION
+
+The session id is retrieved from local storage and recorded. The session ping is submitted.
+
+#### DAEMON
+
+The session id is retrieved from local storage and recorded. The session ping is submitted.
+
+### ON VPN DEACTIVATION
+
+The deactivation signal may not be hit in both the main application and the daemon, because both processes may be force killed. The local storage that contains the session id will be cleared after submitting the final session ping. This will be a flag at initialization that the final session ping was not submitted and there may be lingering data from a previous session in storage that needs to be submitted.
```

Comment:
Correct. I'll try to clarify this.

Reply:
```suggestion
The deactivation signal may not be hit in both the main application and the daemon, because both processes may be force killed. The local storage that contains the session id will be cleared after submitting the final session ping i.e. the ping sent upon VPN deactivation. This will be a flag at initialization that the final session ping was not submitted and there may be lingering data from a previous session in storage that needs to be submitted.
```